### PR TITLE
feat(iam): add new check `iam_policy_cloudshell_admin_not_attached`

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

Add a new check to ensure IAM identities (users, roles, or groups) don't have the AWS managed policy `AWSCloudShellFullAccess` attached. The control fails if this policy is present.

`AWSCloudShellFullAccess` provides full access to AWS CloudShell, including file upload/download capabilities and sudo permissions, which can pose security risks. Adhere to the principle of least privilege and avoid attaching this broad policy.

### Description

Added new check `iam_cloudshell_admin_policy_not_attached` with respective unit tests. Added new attribute `entities_attached_to_cloudshell_policy` with method `_list_entities_for_policy` and a test for the method.

### Checklist

- Are there new checks included in this PR? Yes
    - If so, do we need to update permissions for the provider? No.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/iam/iam_policy_cloudshell_admin_not_attached/iam_policy_cloudshell_admin_not_attached.metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "CheckID": "iam_policy_cloudshell_admin_not_attached",
+  "CheckTitle": "Check if IAM identities (users,groups,roles) have the AWSCloudShellFullAccess policy attached.",
+  "CheckType": [
+    "Software and Configuration Checks/AWS Security Best Practices/CIS AWS Foundations Benchmark"
+  ],
+  "ServiceName": "iam",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:aws:iam::{account-id}:{resource-type}/{resource-id}",
+  "Severity": "medium",
+  "ResourceType": "AwsIamRole, AwsIamUser, AwsIamGroup",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/iam/iam_policy_cloudshell_admin_not_attached/iam_policy_cloudshell_admin_not_attached.metadata.json]:
```suggestion
  "ResourceType": "AwsIamPolicy",
```

File: prowler/providers/aws/services/iam/iam_service.py [hunk 2]
```diff
@@ -685,6 +691,40 @@ def _list_entities_role_for_policy(self, policy_arn):
         finally:
             return roles
 
+    def _list_entities_for_policy(self, policy_arn):
+        logger.info("IAM - List Entities Role For Policy...")
+        try:
+            entities = []
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/iam/iam_service.py]: Please, make `entities` a dictionary with keys as User, Group, Role with a list for each entity.

Comment (MrCloudSec) [reply prowler/providers/aws/services/iam/iam_service.py]: So we can show in the status extended the type of entities too.

File: prowler/providers/aws/services/iam/iam_policy_cloudshell_admin_not_attached/iam_policy_cloudshell_admin_not_attached.metadata.json (hunk 1 above)
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/iam/iam_policy_cloudshell_admin_not_attached/iam_policy_cloudshell_admin_not_attached.metadata.json]: Done -> [e31f73b](https://github.com/prowler-cloud/prowler/pull/5437/commits/e31f73b7edf98318231f4b0043ada613f28ccba6)

File: prowler/providers/aws/services/iam/iam_service.py (hunk 2 above)
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/iam/iam_service.py]: Done -> [1ee24fc](https://github.com/prowler-cloud/prowler/pull/5437/commits/1ee24fc407ed10fc7a10f812a073b597ae811992)

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**